PyInstaller/building/api.py

```diff
@@ -798,7 +799,11 @@ def _set_dependencies(self, analysis, path):
                 else:
                     dep_path = self._get_relative_path(path, self._dependencies[tpl[1]])
                     logger.debug("Referencing %s to be a dependecy for %s, located in %s" % (tpl[1], path, dep_path))
-                    analysis.dependencies.append((":".join((dep_path, tpl[0])), tpl[1], "DEPENDENCY"))
+                    analysis.dependencies.append((
+                        ":".join((dep_path, os.path.basename(tpl[1]))),
```

Comment:
How is this related to "Normalise path casing"?

Reply:
Good question, I don't recall anymore - tpl[0] is the basename of tpl[1] so I'm not sure what the intention was here I've removed this and the tests still work.

Reply:
I recalled it was required, this is due to tpl[0] sometimes having the incorrect casing of the file name, which on operating system which have case-sensitive filepaths (eg Linux) results in the file not being loaded as the path is incorrect. I've added this comment to the comment message.